presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1720,6 +1757,24 @@ else if (criteria instanceof JoinOn) {
                     }
                 }
 
+                if (!checkIfRelationUsedInExpressionJointlyWithOtherRelation(expression, node.getRight())) {
+                    String rightTableName = "";
+                    if (node.getRight() instanceof Table) {
+                        rightTableName = ((Table) node.getRight()).getName().toString();
+                    }
+                    else if (node.getRight() instanceof AliasedRelation) {
```

Comment:
You can just make it an else and instead of make it an assert/checkstate that it has to be an AliasedRelation